### PR TITLE
Add Filter for Formatting Repeating Answers

### PR DESCRIPTION
**Pull request by bitdivision**
### What is the context of this PR?
[Trello](https://trello.com/c/MIDRKUbr/2253-playback-of-a-repeating-answers)

This adds a filter to display lists of items (currently names, in future addresses etc.). These items can be from multiple groups.

### How to review 
- Run `test_repeating_answer_summaries`
- Enter a name
- Ensure that name is displayed on the confirmation question
- Ensure that each of the following confirmation questions also show the current list of names.

- There is an example of the LMS survey which this should 'match' in trello.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: tests/functional/spec/repeating_answer_summaries.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        .getText(RepeatingAnyoneElsePage.displayedDescription()).should.eventually.contain('Carrie Cormorant Court')
+        .click(RepeatingAnyoneElsePage.yes())
+        .click(RepeatingAnyoneElsePage.submit())
+
+        .setValue(RepeatingNamePage.repeatingFirstName(), 'David')
+        .setValue(RepeatingNamePage.repeatingMiddleNames(), 'Dorian')
+        .setValue(RepeatingNamePage.repeatingLastName(), 'Davies')
+        .click(RepeatingNamePage.submit())
+
+        .getText(RepeatingAnyoneElsePage.displayedDescription()).should.eventually.contain('David Dorian Davies');
```
Comment (dwyeradam) [inline tests/functional/spec/repeating_answer_summaries.js]: Shouldn't we check all three people at this point. Not just the latest addition. Same above with Carrie.

**Review by dwyeradam — APPROVED** [review]
Small thing i'd quite like changed but happy to approve.

File: tests/functional/spec/repeating_answer_summaries.spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        .getText(RepeatingAnyoneElsePage.displayedDescription()).should.eventually.contain('Carrie Cormorant Court')
+        .click(RepeatingAnyoneElsePage.yes())
+        .click(RepeatingAnyoneElsePage.submit())
+
+        .setValue(RepeatingNamePage.repeatingFirstName(), 'David')
+        .setValue(RepeatingNamePage.repeatingMiddleNames(), 'Dorian')
+        .setValue(RepeatingNamePage.repeatingLastName(), 'Davies')
+        .click(RepeatingNamePage.submit())
+
+        .getText(RepeatingAnyoneElsePage.displayedDescription()).should.eventually.contain('David Dorian Davies');
```
Comment (dwyeradam) [inline tests/functional/spec/repeating_answer_summaries.spec.js]: We should check all names are still present, not just latest name.

File: tests/app/test_jinja_filters.py [hunk 3]
```diff
@@ -762,3 +763,27 @@ def test_get_answer_label_with_no_answer_label_and_question_titles(self):
 
         # Then
         self.assertEqual(answer_label, 'default_question_title')
+
+
+    @patch('app.jinja_filters.format_unordered_list')
+    def test_format_repeated_summaries_unformatted(self, patched_format): # pylint: disable=no-self-use
```
Comment (jonnyshaw89) [inline tests/app/test_jinja_filters.py]: Can you create another test with a different delimiter eg a comma with an address.
Also test with some empty values to check we don't get `51 my road, , ,Bristol` etc

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/app/test_jinja_filters.py (hunk 3 above)
Comment (bitdivision) [reply tests/app/test_jinja_filters.py]: Good point. Should be sorted.

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by bitdivision** [comment]
@dwyeradam @jonnyshaw89 Ready to merge I think.

**Comment by jonnyshaw89** [comment]
@bitdivision yeh, after the build has finished 👍
